Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -21,48 +21,67 @@ namespace ROS2::Utils
     {
         AZStd::vector<AZStd::string> modifiedLinks;
         using namespace AZ::rapidxml;
-        AZStd::vector<xml_node<>*> links;
+
         for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
         {
-            if (!link->first_node("inertial"))
+            bool modified = false;
+
+            auto inertial = link->first_node("inertial");
+            if (!inertial)
             {
-                links.push_back(link);
+                inertial = urdf->document()->allocate_node(node_element, "inertial");
```

Comment:
I think a warning when we fall back to default values would be helpful in these situations
```suggestion
                AZ_Warning("URDF", false, "Missing inertial tag, applying default tag.");
                inertial = urdf->document()->allocate_node(node_element, "inertial");
```

Reply:
+1 on that